scripts/shared/build_image.sh

```diff
@@ -11,6 +11,7 @@ DEFINE_string 'dockerfile' '' "Dockerfile to build from" 'f'
 DEFINE_string 'buildargs' '' "Build arguments to pass to 'docker build'"
 DEFINE_boolean 'cache' true "Use cached layers from latest image"
 DEFINE_string 'platform' "${default_platform}" 'Platforms to target'
+DEFINE_string 'hash' '' "File to write the hash to" 'h'
```

Comment:
shall we may be using the same hash file always for all images?

Should make it easier to check on runtime later ?

Reply:
This PR writes each image’s hash to the `package/.image....` file which is used to indicate that the image has been built. That way, when we deploy, we can find out (a) which images have been built in the test run and (b) what their hashes are.